scripts/deploy/functions.sh

```diff
@@ -201,6 +201,25 @@ function fail() {
   echo -e "${RED}[FAIL]${ENDCOLOR} $1"
 }
 
+function check_args(){
+  allowed=$2
+
+  # Flag to check if any invalid argument is found
+  invalid_arg=false
+
+  for arg in "$1"; do
+      # Check if the argument is not in the allowed list
+      if [[ ! " ${allowed[@]} " =~ " $arg " ]]; then
+          echo "Invalid argument: $arg"
+          invalid_arg=true
+      fi
+  done
+  if [ "$invalid_arg" = true ]; then
```

Comment:
Nit: use double brackets for consistency with the rest of the file, double equal sign, also with bash you do not need to enclose the variable in double quotes in double brackets:
```suggestion
  if [[ $invalid_arg == true ]]; then
```